docs/proposals/koordlet/20230728-support-cold-memory-compute.md

```diff
@@ -0,0 +1,326 @@
+	idleMemTotalBytes, err2 := koordletutil.GetIdleMemoryTotalKB("/memory/")
+	if err2 != nil {
+		klog.Warningf("failed to collect node idle memory,err", err2)
+	}
+	idlememTotalValue := float64(idleMemTotalBytes)
+	klog.V(4).Infof("collect idle memory value:", idlememTotalValue)
+	//memWithHotPageUsageValue=total-free-idlememTotalValue
+	memTotalKB, _ := koordletutil.GetMemTotalKB()
+	memFreeKB, _ := koordletutil.GetMemFreeKB()
+	memTotalValue := 1024 * float64(memTotalKB)
+	memFreeValue := 1024 * float64(memFreeKB)
+	memWithHotPageUsageValue := memTotalValue - memFreeValue - idlememTotalValue
+	nodeMemorWithHotPageyUsageMetric, err := metriccache.NodeMetricMemoryWithHotPageUsageMetric.GenerateSample(nil, time.Now(), memWithHotPageUsageValue)
+	if err != nil {
+		klog.Warningf("generate node idlememory metrics failed, err %v", err)
+		return nil, err
+	}
+	return nodeMemorWithHotPageyUsageMetric, nil
+}
+
+func (n *nodeResourceCollector) collectNodeResUsed() {
+    if n.IdleMemoryGate {
+		memWithHotPageUsageMetrics, err := n.CollectNodeMemoryWithHotPageUsage(collectTime)
+		if err != nil {
+			klog.Warningf("failed to collect node memory with hot page: ", err)
+			return
+		}
+		nodeMetrics = append(nodeMetrics, memWithHotPageUsageMetrics)
+	}
+}
+```
+
+The same process is executed in podresource collector to collect idle page. Do not repeat.
+
+#### Report memory usage including hot page
+
+![image](../../images/support-cold-memory-2.svg)
+
+collectNodeMetric() is used to query metirc and return CPU And MemUsed in pkg/koordlet/statesinformer/impl/states_nodemetirc.go file.
+
+I want to add hot page compute in this part. After that, collectNodeMetric will return Memused including hot page size.
+
+Add field named MemoryCollectPolicy to represent which method is used to collect memory usage. Such as usageWithHotPageCache, usageWithoutPageCache,  usageWithPageCache.
+
+```go
+// NodeMetricCollectPolicy defines the Metric collection policy
+type NodeMetricCollectPolicy struct {
+	// AggregateDurationSeconds represents the aggregation period in seconds
+	AggregateDurationSeconds *int64 `json:"aggregateDurationSeconds,omitempty"`
+	// ReportIntervalSeconds represents the report period in seconds
+	ReportIntervalSeconds *int64 `json:"reportIntervalSeconds,omitempty"`
+	// NodeAggregatePolicy represents the target grain of node aggregated usage
+	NodeAggregatePolicy *AggregatePolicy `json:"nodeAggregatePolicy,omitempty"`
+	//MemoryCollectPolicy to represent which method is used to collect memory usage. Such as usageWithHotPageCache, usageWithoutPageCache,  usageWithPageCache.
+	MemoryCollectPolicy string `json:"memoryCollectPolicy,omitempty"`
+}
+```
+
+Add code in collectNodeMetric() function. According to nodeMetric.Spec.CollectPolicy.MemoryCollectPolicy, choose memory usage.
+
+
+```go
+func (r *nodeMetricInformer) collectNodeMetric(queryparam metriccache.QueryParam) (corev1.ResourceList, time.Duration, error) {
```

Comment:
We do not need to add detailed code in the proposal (it is not the time to review your code), but we can clarify the workflow and relations of the implementations.